### ISSUE TITLE
SFlux does not have apply(JFlux)

### ISSUE THREAD

**Issue by zackman0010** [reactor/reactor-scala-extensions#32]
The SFlux class does not have an apply method that takes a JFlux. This means that, with the removal of the deprecated Flux, there is no longer a way to convert a JFlux to an SFlux.

**Comment by rs017991** [comment]
@zackman0010 the **ScalaConverters** class adds implicit **asScala** methods to JFlux and JMono, so you can do like:
```
import reactor.core.scala.publisher.ScalaConverters

JFlux.just("whatever").asScala
```

**Comment by sinwe** [comment]
Or another shortcut is
`val sFlux = SFlux.fromPublisher(jFlux)`

**Comment by zackman0010** [comment]
Thank you, I somehow missed that one when I was looking through SFlux last night.

**Comment by zackman0010** [comment]
@sinwe If I submitted a merge request to add another apply method to SFlux that simply calls the fromPublisher method, which is the same thing that SMono's apply does, would it be accepted?

**Comment by sinwe** [comment]
@zackman0010 You can try and raise PR. I'm not sure if that would be working because of below

    object SFlux {
        def apply[T](elements: T*): SFlux[T] = SFlux.fromIterable(elements)
    }

When you do `SFlux(jFlux)` it might be treated as an `SFlux` with elements on single `JFlux` instead of converting the `JFlux` to `SFlux`.
But I might be wrong. That's just on top of my head.

But it'd be nice if it works because that'll make it consistent with `SMono(jMono)`

**Comment by zackman0010** [comment]
You raised a good point, so I went to a scratch sheet to test it out. It looks like, if there's both a T* and a more specific apply function available, it will use the more specific one over the T* one.

![image](https://user-images.githubusercontent.com/9540563/63298376-ef78ce00-c298-11e9-9dbf-a233badf72f5.png)

The test I added on #34 also passed successfully

**Comment by sinwe** [comment]
Thanks... I’ll merge your PR soon and I’ll cut a release soon as we’ve got couple of PRs merged to master

**Comment by sinwe** [comment]
Released on [v0.4.6](https://github.com/reactor/reactor-scala-extensions/releases/tag/0.4.6)
